public/script-tag.js

```diff
@@ -81,14 +79,17 @@
             const cartForm = jQueryPreOrder("form[action='/cart/add']");
             const id = cartForm.serializeArray().find(ele => ele.name === "id").value;
             const addToCartButton = jQueryPreOrder("form[action^='/cart/add']:first [type=submit]:visible:first");
-            const sku = meta.product.variants.find(variant => variant.id == id).sku
+            const sku = meta.product.variants.find(variant => variant.id == id).sku;
 
             const ifPreOrderActive = checkPreOrder(sku);
 
-            if (ifPreOrderActive) {
+            if (ifPreOrderActive.preOrder) {
                 // will add Pre Order to the button
                 addToCartButton.html("Pre Order");
 
+                // will add a span tag to define the date from when the product starts the shipping
+                addToCartButton.after(`<span>Ships from ${ifPreOrderActive.timestamp}</span>`);
```

Comment:
Hi @ymaheshwari1,
We will not add entire text after the button, instead find element with `hc_preordershipsfrom` id and add date to it in human readable format.